### PR TITLE
apidoc links in software-defined assets guide

### PR DESCRIPTION
**Pull request by sryza**
## Checklist

- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: docs/content/guides/dagster/software-defined-assets.mdx [hunk 1]
```diff
@@ -50,17 +50,17 @@ def hottest_dates(daily_temperature_highs: DataFrame) -> DataFrame:
...
 
-`daily_temperature_highs` represents a computed asset. It's derived by taking the `sfo_q2_weather_sample` table and applying the decorated function to it. Notice that it's defined using a pure function - i.e. a function with no side effects, just logical data transformation. The code for storing and retrieving the data in persistent storage will be supplied later on - that allows swapping in different implementations in different environments. E.g. we might want to store data in a local csv file for easy testing, but store data a data warehouse in production.
+`daily_temperature_highs` represents a computed asset. It's derived by taking the `sfo_q2_weather_sample` table and applying the decorated function to it. Notice that it's defined using a pure function - i.e. a function with no side effects, just logical data transformation. The code for storing and retrieving the data in persistent storage will be supplied later on in an <PyObject object="IOManager" /> - that allows swapping in different implementations in different environments. E.g. we might want to store data in a local csv file for easy testing, but store data a data warehouse in production.
```
Comment (clairelin135) [inline docs/content/guides/dagster/software-defined-assets.mdx:55]: Nit: Not related to your change but would be nice to change "csv" -> "CSV"

File: docs/content/guides/dagster/software-defined-assets.mdx [hunk 2]
```diff
@@ -50,17 +50,17 @@ def hottest_dates(daily_temperature_highs: DataFrame) -> DataFrame:
...
 
 `hottest_dates` is a computed asset that depends on another computed asset - the `daily_temperture_highs` asset.
 
 The framework infers asset dependencies by looking at the names of the arguments to the decorated functions. E.g. the function that defines the `daily_temperature_highs` asset has an argument named `sfo_q2_weather_sample` - corresponding to the asset of the same name.
 
 ### Building a job
 
-Having defined some assets, we can build a job that, when it runs, materializes them - i.e. it computes their contents' and writes them to a location in persistent storage.
+Having defined some assets, we can use <PyObject module="dagster.core.asset_defs" object="build_assets_job" /> to build a job that, when it runs, materializes them - i.e. it computes their contents' and writes them to a location in persistent storage.
```
Comment (clairelin135) [inline docs/content/guides/dagster/software-defined-assets.mdx:63]: nit: "contents'" -> "contents"

File: docs/content/guides/dagster/software-defined-assets.mdx [hunk 3]
```diff
@@ -73,9 +73,9 @@ weather_job = build_assets_job(
...
+The order that we supply the assets to <PyObject module="dagster.core.asset_defs" object="build_assets_job" /> doesn't matter. <PyObject module="dagster.core.asset_defs" object="build_assets_job" /> determines the execution dependencies based on the dependencies declared in the assets. `sfo_q2_weather_sample` is included in `source_assets` because it's not computed as part of the job - just consumed by other assets in the job.
 
-The functions we used to define our assets describe how to compute their contents, but now how to read and write them to persistent storage. For reading and writing, we define an `IOManager`. In this case, our `LocalFileSystemIOManager` stores DataFrames as CSVs on the local filesystem:
+The functions we used to define our assets describe how to compute their contents, but now how to read and write them to persistent storage. For reading and writing, we define an <PyObject object="IOManager" />. In this case, our `LocalFileSystemIOManager` stores DataFrames as CSVs on the local filesystem:
```
Comment (clairelin135) [inline docs/content/guides/dagster/software-defined-assets.mdx:78]: nit: "now how" -> "not how"

**Review by clairelin135 — APPROVED** [review]
Thanks for making this update. Caught a couple of typos that are unrelated to your change, but probably a good idea to fix

**Comment by sryza** [comment]
@clairelin135 thanks for catching those typos. I updated the PR with some fixes.

**Comment by clairelin135** [comment]
@sryza LGTM!

**Pull request merged**